functions/index.js

```diff
@@ -1,174 +1,224 @@
-        console.log('got logo image');
-        fsPath = 'media.logoPath';
-    } else if (imageMetadata[type] == 'cover') {
-        console.log('got cover image');
-        fsPath = 'media.coverPath';
-    } else if (imageMetadata[type] == 'wp')  {
-        console.log('got workplace image');
-        fsPath = 'media.wp.' + fileName;
-        value = '';
-    } else {
-        // None handled image type!
-        return;
-    }
-
-    // Update record accordingly
-    var profile = admin.firestore().collection(settingCollection).doc(profileDoc);
-    var data = {};
-    data[fsPath] = value;
-    return profile.update(data);
+function isClient(editor) {
+    return editor === 'AppointmentCreator.client';
 }
 
-// Services image handler
-function servicesImagesHandler(fileObject) {
-    // Do something
+function isBusiness(editor) {
+    return editor === 'AppointmentCreator.business';
 }
 
-// Clients image handler
-function clientsImagesHandler(fileObject) {
-    // Do something
+async function getClientTokens(clientDocID) {
+    const clientResults  = await admin.firestore().collection(clientsCollection).doc(clientDocID).get();
+    const client = clientResults.data();
+    return client['tokens'];
 }
 
-exports.imagesHandler = functions.storage.bucket().object().onFinalize(async (object) => {
-    const filePath = object.name; // File path in the bucket.
+async function getAdminTokens() {
+    let adminTokens = [];
+    const adminResults  = await admin.firestore().collection(adminsCollection).get();
+    for (adminDoc of adminResults.docs) {
+        const adminEmail = adminDoc.id;
+        const adminClientResults = await admin.firestore().collection(clientsCollection).where('email', '==', adminEmail).get();
+        for (adminClientDoc of adminClientResults.docs) {
+            currentTokens = adminClientDoc.data()['tokens'];
+            adminTokens = [...adminTokens, ...currentTokens]
+        }
+    }
+    return adminTokens;
+}
 
-    // Check if file is after resizer
-    if (!filePath.endsWith('_600x600.png')) {
-        // Updating only in case of resized image
-        return;
+// Update the admin when client creates new appointment and update the client when
+// admin creates new appointment for him
+async function HandleNewAppointment(snap, context) {
+    // Getting details of the new document
+    const appointmentId = context.params.appointmentId;
+    const newAppointmentData = snap.data();
+    const clientDocID = newAppointmentData.clientDocID;
+    if ( isClient(newAppointmentData.creator) ) {
+       console.log('Appointment created by client. need to notify the admin');
+       const clientName = newAppointmentData.clientName ?? 'User';
+       const title = clientName + ' Created new appointment';
+       const services = newAppointmentData.services ? newAppointmentData.services.length : 0;
+       const msg = clientName + ' created new appointment for ' +  services + ' services'
```

Comment:
Im not sure we want to show all services in the appointment in the notification body

Reply:
This is just the number of services we need to decide what messages do we want .. i just put it to see that we get the correct data from appointment